cognite/neat/_store/_rules_store.py

```diff
@@ -66,63 +73,153 @@ def calculate_provenance_hash(self, shorten: bool = True) -> str:
+            )
+
+        # Case 4: Provenance is already at the physical state of the data model, going back to logical not possible
+        if not dms and source_entity.dms:
+            raise NeatValueError(
+                "Rules are already in physical state, import of logical model not possible."
+                " Import will be skipped. Start a new NEAT session and import the data model there."
+            )
+
+        # modification took place on information rules
+        if not dms and not source_entity.dms:
+            outside_target_entity = RulesEntity(
+                was_attributed_to=outside_agent,
+                was_generated_by=outside_activity,
+                information=info,
+                dms=dms,
+                issues=issue_list,
+                id_=self._create_id(info, dms),
+            )
+
+        # modification took place on dms rules, keep latest information rules
+        elif dms and source_entity.dms:
+            outside_target_entity = RulesEntity(
+                was_attributed_to=outside_agent,
+                was_generated_by=outside_activity,
+                information=last_change.target_entity.information,
+                dms=dms,
+                issues=issue_list,
+                id_=self._create_id(info, dms),
+            )
+
+        else:
+            raise NeatValueError("Invalid state of rules for manual transformation")
+
+        outside_change = Change(
+            source_entity=last_change.target_entity,
+            agent=outside_agent,
+            activity=outside_activity,
+            target_entity=outside_target_entity,
+            description="Manual transformation of rules outside of NEAT",
+        )
+
+        # record change that took place outside of neat
+        self.provenance.append(outside_change)
+
+        return issue_list
+
+    def import_graph(self, extractor: KnowledgeGraphExtractor) -> IssueList:
+        if not self.empty:
+            raise NeatValueError(f"Data model already exists. Cannot import {extractor.source_uri}.")
+        else:
+            return self.do_activity(partial(self._graph_import_verify_convert, extractor), extractor)
+
+    def import_rules(
         self,
-        action: Callable[[], tuple[InformationRules, DMSRules | None]],
-        agent_tool: BaseImporter | KnowledgeGraphExtractor,
+        importer: BaseImporter,
+        validate: bool = True,
+        client: NeatClient | None = None,
     ) -> IssueList:
-        if self.provenance:
-            raise NeatValueError(f"Data model already exists. Cannot import {agent_tool.source_uri}.")
-        return self.do_activity(action, agent_tool)
+        if self.empty:
+            return self.do_activity(
+                partial(self._rules_import_verify_convert, importer, validate, client),
+                importer,
+            )
+        else:
+            print("Manual transformation of rules is alpha feature. Use with caution.")
```

Comment:
I think we should avoid print in the rule store if we can. Instead, it should be done one the session level

Reply:
I wonder if we should pass en extra argument in read.excel() to enable this feature. Making the user explicit aware that they are using an alpha feature?